torchvision/prototype/transforms/functional/_geometry.py

```diff
@@ -530,9 +533,22 @@ def rotate_bounding_box(
 
     out_bboxes = _affine_bounding_box_xyxy(bounding_box, image_size, angle=-angle, center=center, expand=expand)
 
-    return convert_format_bounding_box(
-        out_bboxes, old_format=features.BoundingBoxFormat.XYXY, new_format=format, copy=False
-    ).view(original_shape)
+    if expand:
+        # The way we recompute image_size is not optimal due to redundant computations of
+        # - rotation matrix (_get_inverse_affine_matrix)
+        # - points dot matrix (_compute_affine_output_size)
+        # Alternatively, we could return new image size by self._F.rotate_bounding_box
+        height, width = image_size
+        rotation_matrix = _get_inverse_affine_matrix([0.0, 0.0], angle, [0.0, 0.0], 1.0, [0.0, 0.0])
+        new_width, new_height = _FT._compute_affine_output_size(rotation_matrix, width, height)
+        image_size = (new_height, new_width)
```

Comment:
@vfdev-5 we should be able to get the `new_width, new_height` out of `_affine_bounding_box_xyxy` without extra / with minimal computation, right?

Reply:
Given that there is no regression in performance here, i.e. I only moved code around, I'll not fix that here but rather open an issue to mark this as a potential performance improvement.

Reply:
Could you please add an in-place performance TODO?